software/web/src/Client/Index.fs

```diff
@@ -356,10 +368,41 @@ let containerBox (model : Model) (dispatch : Msg -> unit) =
             ]
         ]
 
-        Textarea.textarea [
-            Textarea.Value model.Input
-            Textarea.OnChange (fun x -> SetInput x.Value |> dispatch)
-        ] []
+        Level.level [ ][
+            Textarea.textarea [
+                Textarea.Value model.Input
+                Textarea.OnChange (fun x -> SetInput x.Value |> dispatch)
+            ] []
+        ]
+        
+        label [ ] [
+            str "Upload a MIDI file to generate music for the current config."
+        ]
+        input [ 
+            Class "input"
+            Type "file"
+            OnInput (fun ev -> 
+                let file = ev.target?files?(0)
+                let reader = Browser.Dom.FileReader.Create()
+
+                try 
+                    reader?readAsArrayBuffer(file)
+                with
+                    | Failure msg -> System.Console.WriteLine("No file selected. " + msg)
+
+                reader.onload <- fun evt ->
+                    dispatch (SaveMidiFile evt.target?result)
```

Comment:
Should combine the two commands so that updating the config and filename are done at once